v3/src/components/graph/hooks/use-plot.ts

```diff
@@ -43,10 +45,9 @@ export interface IPlotResponderProps {
 
 export const usePlotResponders = (props: IPlotResponderProps) => {
   const {
-      graphModel, primaryAttrID, secondaryAttrID, legendAttrID, enableAnimation,
+      graphModel, dataset, primaryAttrID, secondaryAttrID, legendAttrID, enableAnimation,
```

Comment:
Rather than passing it in, did you try `useDataSetContext()` instead of `graphModel.config.dataset` below? That should be sufficient to trigger a re-render without requiring clients to pass it in.

Reply:
Yes, that worked! Thanks. Cleared up a mystery for me.